Data/Semiring.hs

```diff
@@ -141,10 +141,51 @@ import qualified Prelude as P
 #endif
 
 #if !HOST_OS_WINDOWS
```

Comment:
I believe the `HOST_OS_WINDOWS` part can be removed, it was added in #35 as a bandaid for #34, but this PR should be a proper fix.